cli/command/stack/kubernetes/deploy.go

```diff
@@ -0,0 +1,133 @@
+package kubernetes
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+
+	"github.com/docker/cli/cli/command/stack/options"
+	composeTypes "github.com/docker/cli/cli/compose/types"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// RunDeploy is the kubernetes implementation of docker stack deploy
+func RunDeploy(dockerCli *KubeCli, opts options.Deploy) error {
+	cmdOut := dockerCli.Out()
+	// Check arguments
+	if opts.Composefile == "" {
+		return errors.Errorf("Please specify a Compose file (with --compose-file).")
+	}
+	// Initialize clients
+	stacks, err := dockerCli.stacks()
+	if err != nil {
+		return err
+	}
+	composeClient, err := dockerCli.composeClient()
+	if err != nil {
+		return err
+	}
+	configMaps := composeClient.ConfigMaps()
+	secrets := composeClient.Secrets()
+	services := composeClient.Services()
+	pods := composeClient.Pods()
+	watcher := DeployWatcher{
+		Pods: pods,
+	}
+
+	// Parse the compose file
+	stack, cfg, err := LoadStack(opts.Namespace, opts.Composefile)
+	if err != nil {
+		return err
+	}
+
+	// FIXME(vdemeester) handle warnings server-side
+
+	if err = IsColliding(services, stack); err != nil {
+		return err
+	}
+
+	if err = createFileBasedConfigMaps(stack.Name, cfg.Configs, configMaps); err != nil {
+		return err
+	}
+
+	if err = createFileBasedSecrets(stack.Name, cfg.Secrets, secrets); err != nil {
+		return err
+	}
+
+	if in, err := stacks.Get(stack.Name, metav1.GetOptions{}); err == nil {
```

Comment:
You also may want to rename `in` to `stackObject` or something more explicit.

Reply:
Also the `Get` call might fail for other reasons than the `stack.Name` not being referenced. Worth adding a test on the error value before trying a `stacks.Create`?